pipenv/core.py

```diff
@@ -1574,6 +1548,166 @@ def pip_install(
     return c
 
 
+def pip_install_deps(
+    project,
+    deps=None,
+    allow_global=False,
+    no_deps=False,
+    pre=False,
+    dev=False,
+    selective_upgrade=False,
+    requirements_dir=None,
+    pypi_mirror=None,
+    trusted_hosts=None,
+    use_pep517=True,
+    use_constraint=False,
+):
+    if not trusted_hosts:
+        trusted_hosts = []
+    trusted_hosts.extend(os.environ.get("PIP_TRUSTED_HOSTS", []))
+    if not allow_global:
+        src_dir = os.getenv(
+            "PIP_SRC", os.getenv("PIP_SRC_DIR", project.virtualenv_src_location)
+        )
+    else:
+        src_dir = os.getenv("PIP_SRC", os.getenv("PIP_SRC_DIR"))
+    if not requirements_dir:
+        requirements_dir = vistir.path.create_tracked_tempdir(
+            prefix="pipenv", suffix="requirements"
+        )
+
+    hashed_requirements = tempfile.NamedTemporaryFile(
+        prefix="pipenv-", suffix="-hashed-reqs.txt", dir=requirements_dir, delete=False
+    )
+    not_hashed_requirements = tempfile.NamedTemporaryFile(
+        prefix="pipenv-", suffix="-reqs.txt", dir=requirements_dir, delete=False
+    )
+    for requirement in deps:
+        ignore_hashes = False
+        vcs_or_editable = requirement.is_vcs or requirement.vcs or requirement.editable
+        if vcs_or_editable:
+            ignore_hashes = True
+        # Try installing for each source in project.sources.
+        search_all_sources = project.settings.get("install_search_all_sources", False)
+        if requirement and vcs_or_editable:
+            requirement.index = None
+
+        line = requirement.line_instance.get_line(
+            with_prefix=True,
+            with_hashes=not ignore_hashes,
+            with_markers=True,
+            as_list=False,
+        )
+        sources = get_source_list(
+            project,
+            requirement.index,
+            extra_indexes=list(project.sources),
+            trusted_hosts=trusted_hosts,
+            pypi_mirror=pypi_mirror,
+        )
+        source_names = {src.get("name") for src in sources}
+        if not search_all_sources and requirement.index in source_names:
+            sources = list(filter(lambda d: d.get("name") == requirement.index, sources))
+        if sources and not vcs_or_editable:
```

Comment:
This bit me the first go round because vcs and editable dependencies need to not specify the index, because they won't get resolved.